### PR TITLE
Remove build warnings from Console

### PR DESCRIPTION
**Pull request by anukritijha**
### Type of change

- Refactoring

### Description

- Remove unwanted variables.
- Refactor plan dropdowns and error Loops
- Upgrade React Popper version to 1.3.7
- Fix type mismatches
- Fix Test Case Fails
- Remove Test Warnings

### CODE REVIEW

**Comment by anukritijha** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest

**Comment by k-wall** [comment]
This doesn't look right to me.  For a clean build, I see lots of the following and some test failures. 

[console-init_build_out.txt](https://github.com/EnMasseProject/enmasse/files/4710940/console-init_build_out.txt)


FAIL src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.test.tsx (7.835s)
[INFO]   ● Console
[INFO]
[INFO]     console.error node_modules/react-dom/cjs/react-dom.development.js:88
[INFO]       Warning: An update to AddressSpaceListContainer inside a test was not wrapped in act(...).
[INFO]
[INFO]       When testing, code that causes React state updates should be wrapped into act(...):
[INFO]
[INFO]       act(() => {
[INFO]         /* fire events that update state */
[INFO]       });
[INFO]       /* assert on the output */

**Comment by anukritijha** [comment]
> This doesn't look right to me. For a clean build, I see now lots of the following and some test failures.
> 
> [console-init_build_out.txt](https://github.com/EnMasseProject/enmasse/files/4710940/console-init_build_out.txt)
> 
> FAIL src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.test.tsx (7.835s)
> [INFO] ● Console
> [INFO]
> [INFO] console.error node_modules/react-dom/cjs/react-dom.development.js:88
> [INFO] Warning: An update to AddressSpaceListContainer inside a test was not wrapped in act(...).
> [INFO]
> [INFO] When testing, code that causes React state updates should be wrapped into act(...):
> [INFO]
> [INFO] act(() => {
> [INFO] /* fire events that update state _/ [INFO] }); [INFO] /_ assert on the output */

These fails and test warnings are not because of the changes in this PR. I think it has resulted from a month old PR in the UI test cases. I will create a new JIRA issue to track this issue and fix this.

**Comment by anukritijha** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest

**Comment by anukritijha** [comment]
image

**Comment by anukritijha** [comment]
@k-wall The test cases and warnings are fixed, I had to comment out one test case in AddressSpaceListContainer because the test case needed some refactoring to fix the warning and the it's bound to change entirely in the 2.0 release. So, it seemed more logical at this point to skip it.

File: console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx [hunk 1]
```diff
@@ -139,15 +139,14 @@ export const Configuration: React.FunctionComponent<IConfiguration> = ({
...
-              };
-            }
+            return {
+              value: plan.metadata.name,
+              label: plan.spec.displayName || plan.metadata.name,
+              description:
+                plan.spec.shortDescription || plan.spec.longDescription
+            };
           })
           .filter(plan => plan !== undefined) || [];
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx:151]: Above we have already filter, can we try to add this filter check in above filter and remove this  filter.
addressSpacePlans.filter(plan => plan.spec.addressSpaceType === type)==> addressSpacePlans.filter(plan => plan.spec.addressSpaceType === type && plan !== undefined)

File: console/console-init/ui/src/modules/address-space/dialogs/EditAddressSpaceContainer/EditAddressSpaceContainer.tsx [hunk 2]
```diff
@@ -55,13 +55,12 @@ export const EditAddressSpaceContainer: React.FunctionComponent<{}> = () => {
...
-                value: plan.metadata.name,
-                label: plan.metadata.name
-              };
-            }
+            return {
+              value: plan.metadata.name,
+              label: plan.metadata.name
+            };
           })
           .filter(plan => plan !== undefined) || [];
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/address-space/dialogs/EditAddressSpaceContainer/EditAddressSpaceContainer.tsx:65]: can you try to move this filter check in above filter.
addressSpacePlans.filter(plan => plan.spec.addressSpaceType === addressSpace.type)==>
addressSpacePlans.filter(plan =>plan !== undefined &&  plan.spec.addressSpaceType === addressSpace.type)

File: console/console-init/ui/src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.test.tsx [hunk 3]
```diff
@@ -160,43 +160,42 @@ describe("<AddressSpaceListContainer/>", () => {
...
-    expect(
-      findByText(addressSpaces.addressSpaces[0].metadata.name)
-    ).toBeDefined();
-    expect(
-      findByText(addressSpaces.addressSpaces[0].metadata.namespace)
-    ).toBeDefined();
-  });
+  //TODO: Fix Warning: You seem to have overlapping act() calls
+
+  // it("should render list of spaces", async () => {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.test.tsx:170]: don't need to comment test code , please use xit to skip test.
it==>xit

**Comment by k-wall** [comment]
@anukritijha the test failures are gone.   However, I still see a whole series of build warnings. What is causing these?

```
 make -C console/console-init/
```


[INFO] warning sha.js@2.4.11: Invalid bin entry for "sha.js" (in "sha.js").
[INFO] [3/4] Linking dependencies...
[INFO] warning " > @apollo/react-hooks@3.1.5" has unmet peer dependency "apollo-client@^2.6.4".
[INFO] warning "@apollo/react-hooks > @apollo/react-common@3.1.4" has unmet peer dependency "apollo-client@^2.6.4".
[INFO] warning "@apollo/react-hooks > @apollo/react-common@3.1.4" has unmet peer dependency "apollo-utilities@^1.3.2".
[INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-cache-inmemory@^1.6.2".
[INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-client@^2.6.4".
[INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-link@^1.2.12".
[INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-utilities@^1.3.2".
[INFO] warning " > @patternfly/react-charts@5.3.19" has unmet peer dependency "prop-types@^15.6.1".
[INFO] warning " > @patternfly/react-core@3.153.13" has unmet peer dependency "prop-types@^15.6.1".
[INFO] warning " > @patternfly/react-icons@3.15.16" has unmet peer dependency "prop-types@^15.6.1".
[INFO] warning " > @patternfly/react-table@2.28.39" has unmet peer dependency "prop-types@^15.6.1".
[INFO] warning "@rescripts/cli > @rescripts/utilities@0.0.6" has incorrect peer dependency "react-scripts@^2.1.2".
[INFO] warning " > react-router-last-location@2.0.1" has unmet peer dependency "prop-types@^15.6.0".
[INFO] warning " > babel-loader@8.1.0" has unmet peer dependency "@babel/core@^7.0.0".
[INFO] warning " > babel-loader@8.1.0" has unmet peer dependency "webpack@>=2".
[INFO] warning "@storybook/react > @babel/plugin-transform-react-constant-elements@7.9.0" has unmet peer dependency "@babel/core@^7.0.0-0".
[INFO] warning "@storybook/react > @babel/preset-react@7.9.4" has unmet peer dependency "@babel/core@^7.0.0-0".
[INFO] warning "@storybook/react > @storybook/core > @babel/plugin-proposal-object-rest-spread@7.9.6" has unmet peer dependency "@babel/core@^7.0.0-0".
[INFO] warning "@storybook/react > @storybook/core > @babel/plugin-syntax-dynamic-import@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
[INFO] warning "@storybook/addon-docs > @babel/plugin-transform-react-jsx@7.9.4" has unmet peer dependency "@babel/core@^7.0.0-0".
[INFO] warning "@storybook/addon-actions > @storybook/api@5.3.18" has unmet peer dependency "regenerator-runtime@*".
[INFO] warning " > @storybook/addon-docs@5.3.18" has unmet peer dependency "react-is@^16.8.0".
[INFO] warning "@storybook/addon-docs > @egoist/vue-to-react@1.1.0" has unmet peer dependency "vue@^2.6.10".
[INFO] warning "@storybook/addon-docs > vue-docgen-loader@1.5.0" has unmet peer dependency "webpack@>=4".
[INFO] warning "@storybook/react > @babel/preset-flow@7.9.0" has unmet peer dependency "@babel/core@^7.0.0-0".
[INFO] warning " > @storybook/react@5.3.18" has unmet peer dependency "@babel/core@^7.0.1".
[INFO] warning "@storybook/react > @storybook/core@5.3.18" has unmet peer dependency "@babel/core@*".
[INFO] [4/4] Building fresh packages...
[INFO] info This package requires node-gyp, which is not currently installed. Yarn will attempt to automatically install it. If this fails, you can run "yarn global add node-gyp" to manually install it.
[INFO] [1/4] Resolving packages...
[INFO] warning Error running install script for optional dependency: "/Users/keith/go/src/github.com/enmasseproject/enmasse/console/console-init/ui/node_modules/fsevents: Failed to auto-install node-gyp. Please run \"yarn global add node-gyp\" manually. Error: \"Couldn't find any versions for \\\"node-gyp\\\" that matches \\\"latest\\\" in our cache (possible versions are \\\"\\\"). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.\""
[INFO] warning Error running install script for optional dependency: "/Users/keith/go/src/github.com/enmasseproject/enmasse/console/console-init/ui/node_modules/fsevents: Failed to auto-install node-gyp. Please run \"yarn global add node-gyp\" manually. Error: \"Couldn't find any versions for \\\"node-gyp\\\" that matches \\\"latest\\\" in our cache (possible versions are \\\"\\\"). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.\""info This module is OPTIONAL, you can safely ignore this error
[INFO] Done in 108.66s.

File: console/console-init/ui/src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.test.tsx (hunk 3 above)
Comment (anukritijha) [reply console/console-init/ui/src/modules/address-space/containers/AddressSpaceListContainer/AddressSpaceListContainer.test.tsx]: Done

File: console/console-init/ui/src/modules/address-space/dialogs/EditAddressSpaceContainer/EditAddressSpaceContainer.tsx (hunk 2 above)
Comment (anukritijha) [reply console/console-init/ui/src/modules/address-space/dialogs/EditAddressSpaceContainer/EditAddressSpaceContainer.tsx]: Done

File: console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx (hunk 1 above)
Comment (anukritijha) [reply console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx]: Done

**Comment by anukritijha** [comment]
> @anukritijha the test failures are gone, thanks. However, I still see a whole series of build warnings. What is causing these?
> 
> ```
>  make -C console/console-init/
> ```
> 
> ```
> [INFO] warning sha.js@2.4.11: Invalid bin entry for "sha.js" (in "sha.js").
> [INFO] [3/4] Linking dependencies...
> [INFO] warning " > @apollo/react-hooks@3.1.5" has unmet peer dependency "apollo-client@^2.6.4".
> [INFO] warning "@apollo/react-hooks > @apollo/react-common@3.1.4" has unmet peer dependency "apollo-client@^2.6.4".
> [INFO] warning "@apollo/react-hooks > @apollo/react-common@3.1.4" has unmet peer dependency "apollo-utilities@^1.3.2".
> [INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-cache-inmemory@^1.6.2".
> [INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-client@^2.6.4".
> [INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-link@^1.2.12".
> [INFO] warning " > @apollo/react-testing@3.1.4" has unmet peer dependency "apollo-utilities@^1.3.2".
> [INFO] warning " > @patternfly/react-charts@5.3.19" has unmet peer dependency "prop-types@^15.6.1".
> [INFO] warning " > @patternfly/react-core@3.153.13" has unmet peer dependency "prop-types@^15.6.1".
> [INFO] warning " > @patternfly/react-icons@3.15.16" has unmet peer dependency "prop-types@^15.6.1".
> [INFO] warning " > @patternfly/react-table@2.28.39" has unmet peer dependency "prop-types@^15.6.1".
> [INFO] warning "@rescripts/cli > @rescripts/utilities@0.0.6" has incorrect peer dependency "react-scripts@^2.1.2".
> [INFO] warning " > react-router-last-location@2.0.1" has unmet peer dependency "prop-types@^15.6.0".
> [INFO] warning " > babel-loader@8.1.0" has unmet peer dependency "@babel/core@^7.0.0".
> [INFO] warning " > babel-loader@8.1.0" has unmet peer dependency "webpack@>=2".
> [INFO] warning "@storybook/react > @babel/plugin-transform-react-constant-elements@7.9.0" has unmet peer dependency "@babel/core@^7.0.0-0".
> [INFO] warning "@storybook/react > @babel/preset-react@7.9.4" has unmet peer dependency "@babel/core@^7.0.0-0".
> [INFO] warning "@storybook/react > @storybook/core > @babel/plugin-proposal-object-rest-spread@7.9.6" has unmet peer dependency "@babel/core@^7.0.0-0".
> [INFO] warning "@storybook/react > @storybook/core > @babel/plugin-syntax-dynamic-import@7.8.3" has unmet peer dependency "@babel/core@^7.0.0-0".
> [INFO] warning "@storybook/addon-docs > @babel/plugin-transform-react-jsx@7.9.4" has unmet peer dependency "@babel/core@^7.0.0-0".
> [INFO] warning "@storybook/addon-actions > @storybook/api@5.3.18" has unmet peer dependency "regenerator-runtime@*".
> [INFO] warning " > @storybook/addon-docs@5.3.18" has unmet peer dependency "react-is@^16.8.0".
> [INFO] warning "@storybook/addon-docs > @egoist/vue-to-react@1.1.0" has unmet peer dependency "vue@^2.6.10".
> [INFO] warning "@storybook/addon-docs > vue-docgen-loader@1.5.0" has unmet peer dependency "webpack@>=4".
> [INFO] warning "@storybook/react > @babel/preset-flow@7.9.0" has unmet peer dependency "@babel/core@^7.0.0-0".
> [INFO] warning " > @storybook/react@5.3.18" has unmet peer dependency "@babel/core@^7.0.1".
> [INFO] warning "@storybook/react > @storybook/core@5.3.18" has unmet peer dependency "@babel/core@*".
> [INFO] [4/4] Building fresh packages...
> [INFO] info This package requires node-gyp, which is not currently installed. Yarn will attempt to automatically install it. If this fails, you can run "yarn global add node-gyp" to manually install it.
> [INFO] [1/4] Resolving packages...
> [INFO] warning Error running install script for optional dependency: "/Users/keith/go/src/github.com/enmasseproject/enmasse/console/console-init/ui/node_modules/fsevents: Failed to auto-install node-gyp. Please run \"yarn global add node-gyp\" manually. Error: \"Couldn't find any versions for \\\"node-gyp\\\" that matches \\\"latest\\\" in our cache (possible versions are \\\"\\\"). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.\""
> [INFO] warning Error running install script for optional dependency: "/Users/keith/go/src/github.com/enmasseproject/enmasse/console/console-init/ui/node_modules/fsevents: Failed to auto-install node-gyp. Please run \"yarn global add node-gyp\" manually. Error: \"Couldn't find any versions for \\\"node-gyp\\\" that matches \\\"latest\\\" in our cache (possible versions are \\\"\\\"). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.\""info This module is OPTIONAL, you can safely ignore this error
> [INFO] Done in 108.66s.
> ```

@k-wall The peer dependency warnings happen because the libraries that we use in our app like apollo, patternfly, storybook etc have other dependencies which are not included in our package.json. As these dependencies are a part of node modules, so our application works as expected. But Yarn is being safe here and warning us about these dependencies. A simple solution would just be to add these dependencies in package.json. I have tried to do this in the past, the warnings were gone, however the application fell into another set of issues with Babel which is dependent on a different version. Hence I chose to ignore these warnings. Wdyt?

**Comment by anukritijha** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest

File: console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx (hunk 1 above)
Comment (anukritijha) [reply console/console-init/ui/src/modules/address-space/dialogs/CreateAddressSpace/Configuration.tsx:151]: Had to revert the changes due to test case failure.

**Comment by anukritijha** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest

**Comment by k-wall** [comment]
>  make -C console/console-init/

**Comment by anukritijha** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest.testViewAddressSpacesAsBasicUser

**Comment by kornys** [comment]
@anukritijha you wrongly triggered test, if you want to specify one test from test class you nee to use `#` as separator between test class and test

**Comment by kornys** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest#testViewAddressSpacesAsBasicUser

**Comment by anukritijha** [comment]
> @anukritijha you wrongly triggered test, if you want to specify one test from test class you nee to use `#` as separator between test class and test

Gotcha, thanks for helping out.

**Pull request merged**